src/main/java/com/shilymily/bms/controller/CompanyController.java

```diff
@@ -0,0 +1,46 @@
+package com.shilymily.bms.controller;
+
+import com.shilymily.bms.entity.Company;
+import com.shilymily.bms.service.CompanyService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.ConstraintViolationException;
+import java.util.List;
+
+/**
+ * Created by hmaskai on 9/8/18.
+ */
+@RestController
+@RequestMapping("/master/company")
+public class CompanyController {
+
+    @Autowired
+    private CompanyService companyService;
+
+    @PostMapping(path = "/add", consumes = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.CREATED)
+    public void addOrUpdateCompany(@RequestBody Company company){
+        companyService.addOrUpdateCompany(company);
+    }
+
+    @DeleteMapping(path = "/remove/{name}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void removeCompany(@PathVariable String name){
```

Comment:
Is there a use-case where we still need info regarding removed companies?
Like a backup database or something, or like a status_flag on the company object to see if it is ACTIVE, DELETED etc and we can iterate using the flag.

Reply:
We won't expose this functionality yet. We can think about what to do when we need it.